packages/central-server/src/apiV2/export/constructExportEmail.js

```diff
@@ -25,7 +30,23 @@ ${error}`;
 Note that you need to be logged in to the admin panel for it to work, and after clicking it once, you won't be able to download the file again.`;
 };
 
-export const constructExportEmail = responseBody => ({
-  subject: 'Your export from Tupaia',
-  message: constructMessage(responseBody),
-});
+const generateAttachments = async (responseBody, emailAsAttachment) => {
+  const { filePath } = responseBody;
+  if (emailAsAttachment) {
+    const fileName = path.basename(filePath);
+    const buffer = await fs.readFileSync(filePath);
+    return [{ filename: fileName, content: buffer }];
+  }
+  return [];
+};
+
+export const constructExportEmail = async (responseBody, req) => {
+  const { emailAsAttachment } = req.query;
+  const attachments = await generateAttachments(responseBody, emailAsAttachment);
+  const message = constructMessage(responseBody, emailAsAttachment);
+  return {
+    subject: 'Your export from Tupaia',
+    message,
+    attachments,
+  };
```

Comment:
```suggestion
export const constructExportEmail = async (responseBody, req) => {
  const { emailExportFileMode = 'downloadLink' } = req.query;
  const { error, filePath } = responseBody;
  const subject = 'Your export from Tupaia';
   
  if (error) {
    return {
      subject,
      message: `Unfortunately, your export failed.
${error}`
    }
  }
  
  if (!filePath) {
    throw new Error('No filePath in export response body');
  }
 
  
  if (emailExportFileMode === 'attachment') {
    return {
      subject,
      message: 'Please find your requested export attached to this email.',
      attachments: await generateAttachments(filePath),
    }
  }
  
  const downloadLink = createDownloadLink(filePath);
  return {
    subject,
    message: `Please click this one-time link to download your requested export: ${downloadLink}
Note that you need to be logged in to the admin panel for it to work, and after clicking it once, you won't be able to download the file again.`
  }
```

Decided to try re-writing this to flow a bit more clearly. What do you think? My preferences for this style were:
- Fewer instances of passing branching logic between functions (the `emailExportFileMode` logic now gets handled in one place)
- Not returning the `attachments` property when it's not needed feels a bit cleaner